src/module.c

```diff
@@ -3625,34 +3653,23 @@ int VM_ReplyWithLongDouble(ValkeyModuleCtx *ctx, long double ld) {
  * The command returns VALKEYMODULE_ERR if the format specifiers are invalid
  * or the command name does not belong to a known command. */
 int VM_Replicate(ValkeyModuleCtx *ctx, const char *cmdname, const char *fmt, ...) {
```

Comment:
```suggestion
int VM_Replicate(ValkeyModuleCtx *ctx, const char *cmdname, const char *fmt, ...) {
```

After a fairly lengthy conversation, the core team (and jacob) came up with an alternative proposal. We weren't a fan of the config that effects all modules, but there is an API `voidValkeyModule_SetModuleOptions(ValkeyModuleCtx *ctx, int options);` which allows use to set a specific module config. We can add a new option for `SKIP_COMMAND_VALIDATION`, so specific modules can opt in to it and skip validation. This means that module developers can always use the same Replicate API.

If a module would like to have a per-site decision about validation, they can toggle the SKIP_COMMAND_VALIDATION on/off. 

Sorry for the late input.

Reply:
I will make a change based on above proposal

Reply:
Do we want to let these APIs skip the validation too? because I can see these APIs use `lookupCommandByCString` only for validation purpose
[VM_CreateCommand](https://github.com/valkey-io/valkey/blob/unstable/src/module.c#L1348)
[VM_EmitAOF](https://github.com/valkey-io/valkey/blob/unstable/src/module.c#L7531)

Reply:
For EmitAOF I think we should skip it.
For CreateCommand, I don't think we can skip it, since I believe it's used to prevent duplicates.